Sample_v3/AppDelegate.swift

```diff
@@ -25,6 +26,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         LogConfig.level = .debug
         
         LogStore.registerShared()
+        
+        UIConfig<DefaultExtraData>.default.channelList.avatarView.appearance().backgroundColor = .red
+
```

Comment:
That's the example of customization via `UIAppearance` proxy. Pay attention we don't use `ChannelAvatarView` directly but customize whatever is in `UIConfig` instead. That's the cool feature as end-users can do all the customization by accessing `UIConfig` only.

Reply:
Even though it look interesting, I actually think it's quite dangerous :) look the the following snippet:

```swift
var configA = UIConfig<DefaultExtraData>()
var configB = UIConfig<DefaultExtraData>()

configA.channelList.avatarView.appearance().backgroundColor = .black
configB.channelList.avatarView.appearance().backgroundColor = .white
```
You'd expect, that `configA` will have black background color and `configB` white one, while they both will have white.

Reply:
Yep, but the following would work:

```swift
var configA = UIConfig<DefaultExtraData>()
configA.channelList.avatarView = MineAvatarView.self

var configB = UIConfig<DefaultExtraData>()
configB.channelList.avatarView = YoursAvatarView.self

configA.channelList.avatarView.appearance().backgroundColor = .black
configB.channelList.avatarView.appearance().backgroundColor = .white
```

Using UIConfig for customization the appearance doesn't actually introduce new issues. The one you shown is mostly about `UIAppearance` cuvette

Reply:
If you think writing `configA.channelList.avatarView.appearance()` is simpler than `YoursAvatarView.appearance()` 🤷‍♂️

Reply:
Hah, ok, got your point 😄 

In your solution it will be:

```swift
UIConfig<DefaultExtraData>.default.channelList.avatarView = YoursAvatarView.self
YoursAvatarView.appearance().backgroundColor = .white
```

while mine is:
```swift
UIConfig<DefaultExtraData>.default.channelList.avatarView = MineAvatarView.self
UIConfig<DefaultExtraData>.default.channelList.avatarView.backgroundColor = .white
```

there's no difference but in the former example we don't duplicate the custom avatar view type

Reply:
Anyway, we cannot prevent users from changing the appearance using `UIConfig` but we can get rid of the this debatable сode snippet 😄